### PR TITLE
[CR] Make Power Storage use chest slots

### PR DESCRIPTION
**Pull request by dunstad**
#### Summary

SUMMARY: Interface "Power storage can be uninstalled, uses slots"

#### Purpose of change

This makes the power storage bionics take up 1 slot when bionic slots is enabled, and changes the player display menu and (soon) the bionics menu to display bionics you have multiples of more compactly.

#### Describe the solution

Power storage bionics now take up slots in the torso, and are uninstallable. A flag for bionics that may be installed more than once is added, and the uninstall procedure for bionics is modified to support multiple bionics with the same ID.

If you have two power storage bionics installed, the @ menu will now display them as "2 Power Storage" . The bionics menu now also displays them this way.

#### Describe alternatives you've considered

We could not modify the UI code and have it just print Power Storage over and over, but that seems messy.

The Power Storage bionics could be adjusted to take up more slots if that's necessary. 

#### Additional context

There are 80 slots in the torso, so at the current 1 slot value, if you filled all of those with Power Storage Mk. 1, you'd have 8000 power. If you filled them all with Mk. 2, you'd have 20,000 power.

2_power_storage

uninstall

bionics_menu

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
I'd prefer not installing it "somewhere in the torso", but rather in some bionic specifically designed to contain power storage batteries.

**Comment by Fris0uman** [comment]
What happens when you have multiple bionic with same id and try to remove one, what's displayed in the uninstall menu?

**Comment by dunstad** [comment]
Only one of them is uninstalled, and the menu looks the same as now.

**Comment by SirPendrak** [comment]
"Describe the solution
Power storage bionics now take up slots in the torso, and are uninstallable. " 

Why is that? If you install too many power storages and then you relize you have not enough place for your favourite bionics youre doomed to be walking energy storage without use for eternity? And it makes it impossible to upgrade from MKI's to MKII's in lategame.

**Comment by tinukedaya** [comment]
> "Describe the solution
> Power storage bionics now take up slots in the torso, and are uninstallable. "
> 
> Why is that? If you install too many power storages and then you relize you have not enough place for your favourite bionics youre doomed to be walking energy storage without use for eternity? And it makes it impossible to upgrade from MKI's to MKII's in lategame.

I guess you misunderstood. It means that you CAN indeed uninstall them. So yes, you can free slots later or upgrade to mkII's.

**Comment by kholat** [comment]
Torso should have more slots then imo.

File: src/bionics.cpp [hunk 1]
```diff
@@ -1939,18 +1939,16 @@ int player::get_free_bionics_slots( const body_part bp ) const
...
         debugmsg( "Tried to install bionic %s that is already installed!", b.c_str() );
         return;
     }
 
     units::energy pow_up = bionics[b].capacity;
     mod_max_power_level( pow_up );
-    if( b == "bio_power_storage" || b == "bio_power_storage_mkII" ) {
+    if( pow_up > 0_J ) {
         add_msg_if_player( m_good, _( "Increased storage capacity by %i." ),
                            units::to_kilojoule( pow_up ) );
```
Comment (anothersimulacrum) [inline src/bionics.cpp]: The will not report if the max power wsa increased by an amount less than one joule.
Additionally, the message will show zero if the power was increased by any value less than 1 kilojoule

Comment (dunstad) [reply src/bionics.cpp]: Ok, thanks for letting me know. I'll look at the new menu code to see how they handle power display. I wasn't sure what to use for 0 in the new system.

Comment (anothersimulacrum) [reply src/bionics.cpp]: The message being wrong is technically not your problem to solve (you don't touch it), but it'd be nice if you did fix it.

Comment (anothersimulacrum) [reply src/bionics.cpp]: In fact, I might suggest (if there's no function to format the power like the menu/sidebar do), that you don't do that and then you or someone else makes that function and uses it in the proper places in a separate PR.

Comment (dunstad) [reply src/bionics.cpp]: Changed 0_J to 0_mJ, it looks like that's the smallest unit.

**Pull request closed without merging**